src/prisma/cli/_node.py

```diff
@@ -0,0 +1,320 @@
+            check=check,
+            cwd=cwd,
+            env=env,
+            stdout=stdout,
+            stderr=stderr,
+        )
+
+    @classmethod
+    def resolve(cls, target: Target) -> NodeBinaryStrategy:
+        path = None
+        if config.use_global_node:
+            # TODO: this should check the version
+            path = _get_global_binary(target)
+
+        if path is not None:
+            return NodeBinaryStrategy(path=path, resolver='global')
+
+        return NodeBinaryStrategy.from_nodeenv(target)
+
+    @classmethod
+    def from_nodeenv(cls, target: Target) -> NodeBinaryStrategy:
+        cache_dir = config.nodeenv_cache_dir.absolute()
+        if cache_dir.exists():
+            log.debug(
+                'Skipping nodeenv installation as it already exists at %s',
+                cache_dir,
+            )
+        else:
+            log.debug('Installing nodeenv to %s', cache_dir)
+            subprocess.run(
+                [
+                    sys.executable,
+                    '-m',
+                    'nodeenv',
+                    str(cache_dir),
+                    *config.nodeenv_extra_args,
+                ],
+                check=True,
+                stdout=sys.stdout,
+                stderr=sys.stderr,
+            )
+            if platform.name() == 'windows':
+                cache_dir.joinpath('node').symlink_to('node.exe')
+
+        if not cache_dir.exists():
+            raise RuntimeError(
+                'Could not install nodeenv to the expected directory; See the output above for more details.'
+            )
+
+        # TODO: what hapens on cygwin?
+        if platform.name() == 'windows':
+            bin_dir = cache_dir / 'Scripts'
+            if target == 'node':
+                path = bin_dir / 'node.exe'
+            else:
+                path = bin_dir / f'{target}.cmd'
+        else:
+            path = cache_dir / 'bin' / target
+
+        if target == 'npm':
+            return cls(path=path, resolver='nodeenv')
+        elif target == 'node':
+            return cls(path=path, resolver='nodeenv')
+        else:
+            raise UnknownTargetError(target=target)
+
+
+class NodeJSPythonStrategy(Strategy):
+    target: Target
+    resolver: Literal['nodejs-bin']
+
+    def __init__(self, *, target: Target) -> None:
+        self.target = target
+        self.resolver = 'nodejs-bin'
+
+    def __run__(
+        self,
+        *args: str,
+        check: bool = False,
+        cwd: Path | None = None,
+        stdout: File | None = None,
+        stderr: File | None = None,
+        env: Mapping[str, str] | None = None,
+    ) -> subprocess.CompletedProcess[bytes]:
+        func = None
+        if self.target == 'node':
+            func = nodejs.node.run
+        elif self.target == 'npm':
+            func = nodejs.npm.run
+        else:
+            raise UnknownTargetError(target=self.target)
+
+        # TODO
+        return func(  #  type: ignore
```

Comment:
We have logging of the invocation in the `NodeBinaryStrategy` -- I reckon we should add that debug output here as well before we make the function call to ease debugging